core/storage/btree.rs

```diff
@@ -312,20 +367,42 @@ impl BTreeCursor {
                             "we don't iterate leaf cells while trying to move to a leaf cell"
                         );
                     }
-                    BTreeCell::IndexInteriorCell(_) => {
-                        unimplemented!();
+                    BTreeCell::IndexInteriorCell(IndexInteriorCell {
+                        left_child_page,
+                        payload,
+                        ..
+                    }) => {
+                        let SeekKey::IndexKey(index_key) = key else {
+                            unreachable!("index seek key should be a record");
+                        };
+                        mem_page.advance();
+                        let record = crate::storage::sqlite3_ondisk::read_record(payload)?;
+                        let comparison = match cmp {
+                            SeekOp::GT => index_key <= &record,
+                            SeekOp::GE => index_key < &record,
+                            SeekOp::EQ => index_key < &record,
```

Comment:
This too

Reply:
I think here we indeed need to change some things because index interior cells contain data too

In fact theres a bug when scanning through an index (empty rows in between):

```
limbo> select age from users where age >= 100 limit 20;
100
100
100
100
100
100
100
100

100
100

100
100
100
100
100
100
100
100
```

Not seeing this same bug with primary key index scans

Reply:
Explained above